src/Sulu/Bundle/MediaBundle/Media/Storage/AzureBlobStorage.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Media\Storage;
+
+use League\Flysystem\AzureBlobStorage\AzureBlobStorageAdapter;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemInterface;
+use MicrosoftAzure\Storage\Blob\BlobRestProxy;
+
+class AzureBlobStorage extends FlysystemStorage
+{
+    /**
+     * @var AzureBlobStorageAdapter
+     */
+    private $adapter;
+
+    /**
+     * @var string
+     */
+    private $container;
+
+    /**
+     * @var BlobRestProxy
+     */
+    private $client;
+
+    public function __construct(
+        FilesystemInterface $filesystem,
+        BlobRestProxy $client,
+        string $container,
+        int $segments
+    ) {
+        parent::__construct($filesystem, $segments);
+
+        if (!$filesystem instanceof Filesystem || !$filesystem->getAdapter() instanceof AzureBlobStorageAdapter) {
+            throw new \RuntimeException();
+        }
+
+        $this->client = $client;
+        $this->container = $container;
+        $this->adapter = $filesystem->getAdapter();
+    }
+
+    public function load(array $storageOptions)
+    {
+        $resource = parent::load($storageOptions);
+        $contents = stream_get_contents($resource);
+
+        $stream = fopen('php://memory', 'r+');
```

Comment:
@plozmun @alexander-schranz why do have to copy it to the memory?

Reply:
Azure Filesystem returns a not seekable object, not like Google Filesystem than returns a PSR7Message seekable, then this fails when it tries to generate thumbnail https://github.com/sulu/sulu/blob/master/src/Sulu/Bundle/MediaBundle/Media/ImageConverter/ImagineImageConverter.php#L152
due to this https://github.com/sulu/sulu/blob/master/src/Sulu/Bundle/MediaBundle/Media/ImageConverter/MediaImageExtractor.php#L52 mime_content_type doest't support not seekable objects

Reply:
OK sounds logic to me - i would at least add a comment for this to be sure that we do not remove it intentionally.

Reply:
Done! Check the comment and tell me if it's enough

Reply:
Can we add the comment before `$contents = stream_get_contents($resource);` on the method its mostly overlooked.

Would also add more information to it e.g.:

```php
// Azure Filesystem returns a not seekable resource
// Need convert to a seekable resource to allow get mimetype from it for MediaImageExtractor
$contents = stream_get_contents($resource);
...
```

Reply:
@plozmun thank you!